src/app/SubscriptionResumptionStorage.h

```diff
@@ -0,0 +1,162 @@
+        SubscriptionId mSubscriptionId;
+        uint16_t mMinInterval;
+        uint16_t mMaxInterval;
+        bool mFabricFiltered;
+        Platform::ScopedMemoryBufferWithSize<AttributePathParamsValues> mAttributePaths;
+        Platform::ScopedMemoryBufferWithSize<EventPathParamsValues> mEventPaths;
+        CHIP_ERROR SetAttributePaths(const ObjectList<AttributePathParams> * pAttributePathList)
+        {
+            mAttributePaths.Free();
+            const ObjectList<AttributePathParams> * attributePath = pAttributePathList;
+            size_t attributePathCount                             = 0;
+            while (attributePath)
+            {
+                attributePathCount++;
+                attributePath = attributePath->mpNext;
+            }
+            ReturnErrorCodeIf((attributePathCount * sizeof(AttributePathParamsValues)) > UINT16_MAX, CHIP_ERROR_NO_MEMORY);
+            mAttributePaths.Calloc(attributePathCount);
+            ReturnErrorCodeIf(mAttributePaths.Get() == nullptr, CHIP_ERROR_NO_MEMORY);
+            attributePath = pAttributePathList;
+            for (size_t i = 0; i < attributePathCount; i++)
+            {
+                mAttributePaths[i].SetValues(attributePath->mValue);
+                attributePath = attributePath->mpNext;
+            }
+            return CHIP_NO_ERROR;
+        }
+        CHIP_ERROR SetEventPaths(const ObjectList<EventPathParams> * pEventPathList)
+        {
+            mEventPaths.Free();
+            const ObjectList<EventPathParams> * eventPath = pEventPathList;
+            size_t eventPathCount                         = 0;
+            while (eventPath)
+            {
+                eventPathCount++;
+                eventPath = eventPath->mpNext;
+            }
+            ReturnErrorCodeIf((eventPathCount * sizeof(EventPathParamsValues)) > UINT16_MAX, CHIP_ERROR_NO_MEMORY);
+            mEventPaths.Calloc(eventPathCount);
+            ReturnErrorCodeIf(mEventPaths.Get() == nullptr, CHIP_ERROR_NO_MEMORY);
+            eventPath = pEventPathList;
+            for (size_t i = 0; i < eventPathCount; i++)
+            {
+                mEventPaths[i].SetValues(eventPath->mValue);
+                eventPath = eventPath->mpNext;
+            }
+            return CHIP_NO_ERROR;
+        }
+    };
+
+    using SubscriptionInfoIterator = CommonIterator<SubscriptionInfo>;
+
+    virtual ~SubscriptionResumptionStorage(){};
+
+    /**
+     * Iterate through persisted subscriptions
+     *
+     * @return A valid iterator on success. Use CommonIterator accessor to retrieve SubscriptionInfo
+     */
+    virtual SubscriptionInfoIterator * IterateSubscriptions() = 0;
+
+    /**
+     * Save subscription resumption information to storage.
+     *
+     * @param subscriptionInfo the subscription information to save - caller should expect the passed in value is consumed
+     * @return CHIP_NO_ERROR on success, else an appropriate CHIP error on failure
```

Comment:
nit: This `@return` description does not help reading the code. It seems obvious that CHIP_ERROR means returning no error on success and some error on failure.

Reply:
Comment applies throughout: if there is something non-obvious or helpful to report, please comment it. Otherwise, we can save some vertical space.

Reply:
Alright I can remove some obvious comments